server/src/inmantals/server.py

```diff
@@ -76,110 +79,145 @@ class Folder:
     :param handler: reference to the InmantaLSHandler responsible for this folder
     """
 
-    folder_uri: URI
+    folder_uri: str
     name: str
     inmanta_project_dir: Union[Optional[tempfile.TemporaryDirectory], str]
     handler: "InmantaLSHandler"
 
-    def __init__(self, folder_uri: URI, name: str, handler: "InmantaLSHandler"):
-        self.folder_uri = folder_uri
-        self.name = name
+    def __init__(self, workspace_folder: lsp_types.WorkspaceFolder, handler: "InmantaLSHandler"):
+        folder_uri = urlparse(workspace_folder["uri"])
```

Comment:
I don't think so, because of [this](https://github.com/inmanta/vscode-inmanta/blob/aded8301d20e70376857c0577a4c7ec84fb55650/src/language_server.ts#L225) the language server should only react to files on disk.

Reply:
Cool, then we could even use pydantic's `FileUrl` type for validation.